lib/encoder.js

```diff
@@ -906,6 +911,9 @@ Encoder.prototype.encodeMap = function (value, subtypes) {
     //include item
```

Comment:
Object can not contain null keys but looking into ES6 `Map`, it looks like it can contain a null/undefined key :open_mouth: , this is valid:

```javascript
var m = new Map();
m.set(1, 1);
m.set(null, 2);
```
Will fix